src/sentry/incidents/typings/metric_detector.py

```diff
@@ -93,3 +98,71 @@ def from_action_model(cls, action: Action) -> NotificationContext:
             target_identifier=action.config.get("target_identifier"),
             target_display=action.config.get("target_display"),
         )
+
+
+@dataclass
+class MetricIssueContext:
+    open_period_identifier: int
+    snuba_query: SnubaQuery
+    new_status: IncidentStatus
+    metric_value: float | None
+
+    @classmethod
+    def _get_incident_status(cls, group: Group, occurrence: IssueOccurrence) -> IncidentStatus:
+        if group.status == GroupStatus.RESOLVED:
+            return IncidentStatus.CLOSED
+        elif occurrence.initial_issue_priority == PriorityLevel.MEDIUM.value:
+            return IncidentStatus.WARNING
+        else:
+            return IncidentStatus.CRITICAL
+
+    @classmethod
+    def _get_snuba_query(cls, occurrence: IssueOccurrence) -> SnubaQuery:
+        snuba_query_id = occurrence.evidence_data.get("snuba_query_id")
+        if not snuba_query_id:
+            raise ValueError("Snuba query ID is required for alert context")
+        try:
+            query = SnubaQuery.objects.get(id=snuba_query_id)
+        except SnubaQuery.DoesNotExist as e:
+            raise ValueError("Snuba query does not exist") from e
+        return query
+
+    @classmethod
+    def _get_new_status(cls, group: Group, occurrence: IssueOccurrence) -> IncidentStatus:
+        return cls._get_incident_status(group, occurrence)
```

Comment:
What's the purpose of this method if it's just a thin pass-through for `_get_incident_status`.